src/io.rs

```diff
@@ -403,10 +403,17 @@ pub fn build_input() -> (Stream, StreamConfig, Receiver<AudioBuffer>) {
     let mut supported_configs_range = device
         .supported_input_configs()
         .expect("error while querying configs");
-    let supported_config = supported_configs_range
-        .next()
-        .expect("no supported config?!")
-        .with_max_sample_rate();
+
+    let supported_config = match options.sample_rate {
+        Some(sample_rate)=> supported_configs_range
+            .next()
+            .expect("no supported config?!")
+            .with_sample_rate(CpalSampleRate(sample_rate)),
```

Comment:
Requesting a specific sample rate should be done after determining the fallback config.
The fallback config is put in the variable `default_config` and should only contain values that were actually supplied by cpal, so they should never fail.
Could you try to set the requested sample rate only on the `config` settings?